packages/@sanity/components/src/dialogs/PanePopover.js

```diff
@@ -0,0 +1,75 @@
+/* eslint-disable complexity */
+import React from 'react'
+import PropTypes from 'prop-types'
+import CheckCircleIcon from 'part:@sanity/base/circle-check-icon'
+import WarningIcon from 'part:@sanity/base/warning-icon'
+import ErrorIcon from 'part:@sanity/base/error-icon'
+import InfoIcon from 'part:@sanity/base/info-icon'
+import classNames from 'classnames'
+import styles from './styles/PanePopover.css'
+
+export default class PanePopover extends React.PureComponent {
+  static propTypes = {
+    children: PropTypes.node,
+    icon: PropTypes.oneOfType([PropTypes.string, PropTypes.node, PropTypes.bool]),
+    kind: PropTypes.oneOf(['info', 'warning', 'error', 'success', 'neutral']),
+    title: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+    subtitle: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+    id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired
+  }
+
+  static defaultProps = {
+    children: null,
+    icon: null,
+    kind: 'info',
+    title: 'Pane Popover',
+    subtitle: 'Popover content'
+  }
+
+  DEFAULT_ICONS = {
+    info: <InfoIcon />,
+    success: <CheckCircleIcon />,
+    warning: <WarningIcon />,
+    error: <WarningIcon />
+  }
+
+  snackIcon = () => {
+    const {icon, kind} = this.props
+    if (typeof icon === 'boolean' && icon) return this.DEFAULT_ICONS[kind]
+    if (typeof icon === 'object' || typeof icon === 'string') return icon
```

Comment:
What is this case intended for? Strings are icons? Object?
I would personally just make the `DEFAULT_ICONS` be the non-rendered versions, eg `info: InfoIcon`.

Reply:
The initial idea was for the snackbar, that strings and icons were both valid icon types depending on what someone would want to use in their snackbar. I guess this doesn't apply here!